File: API/Entities/Invoice.cs
namespace  API.Entities;

public class Invoice
{
    public int Id { get; set; }
    public int WorkOrderId { get; set; }
    public int ContractorId { get; set; }
    public decimal TotalAmount { get; set; }
    public ICollection<InvoiceItem> InvoiceItems { get; set; }
    public DateTime? DueDate { get; set; }
    public DateTime? DatePaid { get; set; }
    public WorkOrder WorkOrder { get; set; }
    public AppUser Contractor { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.Now;
    public bool Deleted { get; set; }
}

File: client/src/app/_models/invoice.ts
import { InvoiceItem } from "./invoiceItem";
import { Member } from "./member";
import { WorkOrder } from "./workorder";

export interface Invoice {
  id: number;
  workorderId: number;
  workorder: WorkOrder;
  contractorId: number;
  contractor: Member;
  amount: number;
  createdDate: Date;
  dueDate?: Date;
  datePaid?: Date;
  invoiceItems: InvoiceItem[];
}

